bin/alpen-cli/src/recovery.rs

```diff
@@ -248,3 +253,83 @@ pub struct InvalidPrivateKey(OneOf<(FromUtf8Error, DescriptorKeyParseError)>);
 #[derive(Debug)]
 #[allow(unused)]
 pub struct InvalidPublicKey(OneOf<(FromUtf8Error, DescriptorKeyParseError)>);
+
+#[derive(Debug)]
+pub struct DescriptorRecoveryKey {
+    pub recover_at: u32,
+    pub desc_string_hash: [u8; 32],
+}
+
+impl DescriptorRecoveryKey {
+    pub fn new(recover_at: u32, desc: &Descriptor<DescriptorPublicKey>) -> Self {
+        let mut hasher = <Sha256 as Digest>::new(); // this is to appease the analyzer
+        hasher.update(desc.to_string().as_bytes());
+        Self {
+            recover_at,
+            desc_string_hash: hasher.finalize().into(),
+        }
+    }
+
+    pub fn encode(&self) -> [u8; 36] {
+        make_buf! {
+            (&self.recover_at.to_be_bytes(), 4),
+            (&self.desc_string_hash, 32)
+        }
+    }
+
+    pub fn decode(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 36 {
+            return None;
+        }
+
+        let recover_at = u32::from_be_bytes(unsafe { *(bytes[..4].as_ptr() as *const [_; 4]) });
+        let mut desc_string_hash = [0u8; 32];
+        desc_string_hash.copy_from_slice(&bytes[4..36]);
```

Comment:
Every time that I see magic numbers in array slicing I am hugely concerned with code maintainability, especially with `unsafe` in the mix.

Despite the why of `4` and `32` being in eyesight for the reader (you can see why these magic numbers have their value by seeing slight above), I think we should move these to module-level `const`s. They can be non-`pub` for now.